azbus/sender.go

```diff
@@ -150,3 +150,56 @@ func (s *Sender) Send(ctx context.Context, message *OutMessage) error {
 	log.Debugf("Sending message id %s took %s", id, time.Since(now))
 	return nil
 }
+
+func (s *Sender) NewMessageBatch(ctx context.Context) (*OutMessageBatch, error) {
+	return s.sender.NewMessageBatch(ctx, nil)
+}
+
+// BatchAddMessage calls Addmessage on batch
+// Note: this method is a direct pass through and exists only to provide a
+// mockable interface for adding messages to a batch.
+func (s *Sender) BatchAddMessage(batch *OutMessageBatch, m *OutMessage, options *azservicebus.AddMessageOptions) error {
+	return batch.AddMessage(m, options)
+}
+
+// SendBatch submits a message batch to the broker. Ignores cancellation.
+func (s *Sender) SendBatch(ctx context.Context, batch *OutMessageBatch) error {
+
+	// Without this fix eventsourcepoller and similar services repeatedly context cancel and repeatedly
+	// restart.
+	ctx = context.WithoutCancel(ctx)
```

Comment:
is eventsourcepoller actually using this?

Reply:
uh that must be a copy paste from the existing code. we don't have an eventsourcepoller any more. not sure